coffee-chats/src/main/webapp/src/components/MultiDatePicker.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactElement, MouseEvent } from "react";
+import "./MultiDatePicker.css"
+import { DatePicker } from "@material-ui/pickers"
+import { isSameDay } from "date-fns";
+import { MaterialUiPickersDate } from "@material-ui/pickers/typings/date";
+
+interface MultiDatePickerProps {
+  dates: Date[],
+  setDates: (dates: Date[]) => void,
+}
+
+
+export const MultiDatePicker: React.FC<MultiDatePickerProps> = ({dates, setDates}) => {
+
+  // Custom rendering function for day Element on calendar
+  const renderDay = (day: MaterialUiPickersDate, selectedDate: MaterialUiPickersDate, dayInCurrentMonth: boolean, dayComponent: ReactElement) => {
+    return React.cloneElement(dayComponent, {
+      onClick: (e: MouseEvent) => {
+        e.stopPropagation();
+        const foundIndex = dates.findIndex(d => isSameDay(d, day as Date));
+
+        // Toggle selection of day appearing in selected dates array
+        if (foundIndex >= 0) {
+          const newDates = [...dates]
+          newDates.splice(foundIndex, 1)
+          setDates(newDates)
+        } else {
+          setDates([...dates, day as Date])
+        }
+      },
+
+      selected: dates.find(d => isSameDay(d, day as Date)) != null
+    });
+  }
+
+  return (
+    <div className="datePicker">
+      <DatePicker
+        disablePast
+        disableToolbar
+        value={dates[0]}
+        variant="static"
+        renderDay={renderDay}
+        onChange={() => {}}
```

Comment:
Does it do something else by default?

Reply:
The Picker is usually intended to just store a single date, so the default onChange resets the value to the newly selected one. In order to select multiple days, I have moved the 'onClick' functionality to the day elements themselves.

Reply:
How that would work for end of the month, if I want to select 31 of Aug and 1 of Sept?

Reply:
You are able to select any dates you like, and can click to the next month (with the arrows at the top of the calendar), if you would like to do so. I am not sure I entirely follow.